apps/champions/test/battle_test.exs

```diff
@@ -680,6 +680,158 @@ defmodule Champions.Test.BattleTest do
       assert "team_1" ==
                Champions.Battle.Simulator.run_battle([unit], [target_dummy_5], maximum_steps: maximum_steps).result
     end
+
+    test "Self Heal" do
```

Comment:
Let's add an explanation of the battle we're running here so it's easier to follow! Check Execution-Heal for an example. It doesn't need to be super detailed since the test is quite simple, but some small explanation goes a long way!